src/posposet/common_test.go

```diff
@@ -0,0 +1,308 @@
+package posposet
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/Fantom-foundation/go-lachesis/src/common"
+)
+
+func TestParseEvents(t *testing.T) {
+	assert := assert.New(t)
+
+	nodes, events, names := ParseEvents(`
+a00 b00   c00 d00
+║   ║     ║   ║
+a01 ║     ║   ║
+║   ╠  ─  c01 ║
+a02 ╣     ║   ║
+║   ║     ║   ║
+╠ ─ ╫ ─ ─ c02 ║
+║   b01  ╝║   ║
+║   ╠ ─ ─ ╫ ─ d01
+║   ║     ║   ║
+║   ║     ║   ║
+╠ ═ b02 ═ ╬   ╣
+║   ║     ║  ║║
+a03 ╣     ╠ ─ d02
+║║  ║     ║  ║║
+║║  ║     ║  ║╠ ─ e00
+║║  ║     ║   ║   ║
+a04 ╫ ─ ─ ╬  ╝║   ║
+║║  ║     ║   ║   ║
+║╚  ╫╩  ─ c03 ╣   ║
+║   ║     ║   ║   ║
+`)
+	expected := map[string][]string{
+		"a00": {""},
+		"a01": {"a00"},
+		"a02": {"a01", "b00"},
+		"a03": {"a02", "b02"},
+		"a04": {"a03", "c02", "d01"},
+		"b00": {""},
+		"b01": {"b00", "c01"},
+		"b02": {"b01", "a02", "c02", "d01"},
+		"c00": {""},
+		"c01": {"c00", "b00"},
+		"c02": {"c01", "a02"},
+		"c03": {"c02", "a03", "b01", "d02"},
+		"d00": {""},
+		"d01": {"d00", "b01"},
+		"d02": {"d01", "c02"},
+		"e00": {"", "d02"},
+	}
+
+	if !assert.Equal(5, len(nodes), "node count") {
+		return
+	}
+	if !assert.Equal(len(expected), len(names), "event count") {
+		return
+	}
+
+	index := make(map[EventHash]*Event)
+	for _, nodeEvents := range events {
+		for _, e := range nodeEvents {
+			index[e.Hash()] = e
+		}
+	}
+
+	for eName, e := range names {
+		parents := expected[eName]
+		if !assert.Equal(len(parents), len(e.Parents), "at event "+eName) {
+			return
+		}
+		for _, pName := range parents {
+			hash := ZeroEventHash
+			if pName != "" {
+				hash = names[pName].Hash()
+			}
+			if !e.Parents.Contains(hash) {
+				t.Fatalf("%s has no parent %s", eName, pName)
+			}
+		}
+	}
+}
+
+/*
+ * Utils:
+ */
+
+// FakePoset creates empty poset with mem store and equal stakes of nodes in genesis.
+func FakePoset(nodes []common.Address) *Poset {
+	balances := make(map[common.Address]uint64, len(nodes))
+	for _, addr := range nodes {
+		balances[addr] = uint64(1)
+	}
+
+	store := NewMemStore()
+	err := store.ApplyGenesis(balances)
+	if err != nil {
+		panic(err)
+	}
+
+	p := New(store)
+	return p
+}
+
+// ParseEvents parses events from ASCII-scheme.
+// Use joiners ║ ╬ ╠ ╣ ╫ ╚ ╝ ╩ and optional fillers ─ ═ to draw ASCII-scheme.
```

Comment:
Great idea, but better to keep it in lower ASCII range to be proof to charset conversions.

Reply:
There isn't such pretty pseudographics. Which symbols from upper ASCII range do you advise?

Reply:
lower ASCII range, which is invariant for most popular charsets in use, not the upper.

\|/+- are traditionally used in ASCII graph pseudo graphics, we also can use *" if needed.
It is not about being pretty, just functional and be proof to charset conversion.

Reply:
Actually the idea doesn't look as good as for the first sight, it would be mundane to draw any complex graph and it would be also prone to auto reformating.
We better to integrate something similar to DOT format (https://en.wikipedia.org/wiki/DOT_(graph_description_language) ), which could be easily drawn in any tool graph visualisation for humans.

Reply:
DOT format would be overengineering here. My local decision is for poset tests and inplace reading only. 
- is it better than [playlist](https://github.com/Fantom-foundation/go-lachesis/blob/95e43ed2e6c6532b2bf4be8e15ca16b6785eeb25/src/poset/poset_test.go#L2505)? Of cause! So, it is worthy to be approved;
- can you make better? Of cause! So, you are wellcome to present your own PR;

Reply:
It is not better than playlist, it is just fancier. Playlist allows to create events with payload while you approach doesn't. Playlist is proof to character conversion, while your solution isn't. Playlist is proof to (auto)reformating, - your isn't. Plus creation of relatively complex graphs would require more typing with your approach.

Reply:
Actually don't worry about this, Go enforces UTF-8, even a simple `go build` will change any `.go` file with a different encoding to UTF-8.

Reply:
@devintegral Was just chatting with @Maxime2, so you're going from pseudographics to Go structures. That's good. Will also need its inverse implemented, going from Go structures back to pseudographics. Just because it's hard to write these pseudographics by hand.

Reply:
@SamuelMarks, it's easy, but when we need to go from structures back to pseudographics?